core/src/main/scala/kanaloa/dispatcher/Dispatcher.scala

```diff
@@ -22,7 +22,9 @@ trait Dispatcher extends Actor with ActorLogging {
   def settings: Dispatcher.Settings
   def backend: Backend
   def resultChecker: ResultChecker
-  def metricsCollector: ActorRef
+  def reporter: Option[Reporter]
+
+  protected lazy val metricsCollector = context.actorOf(MetricsCollector.props(reporter, settings.performanceSamplerSettings))
```

Comment:
It does send messages directly to it, via:
```scala
case SubscribePerformanceMetrics(actor)      ⇒ metricsCollector ! PerformanceSampler.Subscribe(actor)
    case UnSubscribePerformanceMetrics(actor)    ⇒ metricsCollector ! PerformanceSampler.Unsubscribe(actor)
```.

Not the largest deal in the world, but I was just pointing out that this coupling could potentially make testing difficult.  In either event, if that DOES happen, changing this to be injected is pretty easy.